File: Sesion 6/Marco teorico/7CÓMO SE DECLARAN LAS CLASES AMIGAS/7.py
class MiClase:
    def __init__(self, dato):
        self.__dato_privado = dato

class ClaseAmiga:
    def mostrar_dato_privado(self, obj):
        print("El dato privado de MiClase es:", obj._MiClase__dato_privado)


objeto = MiClase(42)
amigo = ClaseAmiga()
amigo.mostrar_dato_privado(objeto)

File: Sesion 7/Marco teorico/7forma_de_uso/7.java
class MiClase {
    private int datoPrivado;

    public MiClase(int dato) {
        this.datoPrivado = dato;
    }

    public int getDatoPrivado() {
        return datoPrivado;
    }
}

class ClaseAmiga {
    public void mostrarDatoPrivado(MiClase obj) {
        System.out.println("El dato privado de MiClase es: " + obj.getDatoPrivado());
    }
}

public class Main {
    public static void main(String[] args) {
        MiClase objeto = new MiClase(42);
        ClaseAmiga amigo = new ClaseAmiga();
        amigo.mostrarDatoPrivado(objeto);
    }
}
